scipy/ndimage/_backend.py

```diff
@@ -0,0 +1,266 @@
+    >>> correlate(np.arange(10), np.array([1, 2.5]))
+    array([ 0,  2,  6,  9, 13, 16, 20, 23, 27, 30])
+    >>> set_global_backend("scipy")  # Restore global backend to default
+
+    """
+    backend = _backend_from_arg(backend)
+    ua.register_backend(backend)
+
+
+def set_backend(backend, coerce=True, only=False):
+    """
+    Context manager to set the backend within a fixed scope.
+
+    Upon entering the ``with`` statement, the given backend will be added to
+    the list of available backends with the highest priority. Upon exit, the
+    backend is reset to the state before entering the scope.
+
+    Parameters
+    ----------
+    backend : {object, 'scipy'}
+        The backend to use.
+        Can either be a ``str`` containing the name of a known backend
+        {'scipy'} or an object that implements the uarray protocol.
+    coerce : bool, optional
+        Whether to allow expensive conversions for the ``x`` parameter. e.g.,
+        copying a NumPy array to the GPU for a CuPy backend. Implies ``only``.
+    only : bool, optional
+        If only is ``True`` and this backend returns ``NotImplemented``, then a
+        BackendNotImplemented error will be raised immediately. Ignoring any
+        lower priority backends.
+
+    Notes
+    -----
+    .. versionadded:: 1.8.0
+
+    Examples
+    --------
+    >>> import scipy.ndimage as ndimage
+    >>> with ndimage.set_backend('scipy', only=True):
+    ...     # Always calls the scipy implementation
+    ...     ndimage.correlate(np.arange(10), [1, 2.5])
+    array([ 0,  2,  6,  9, 13, 16, 20, 23, 27, 30])
+
+    """
+    backend = _backend_from_arg(backend)
+    return ua.set_backend(backend, coerce=coerce, only=only)
+
+
+def skip_backend(backend):
+    """Context manager to skip a backend within a fixed scope.
+
+    Within the context of a ``with`` statement, the given backend will not be
+    called. This covers backends registered both locally and globally. Upon
+    exit, the backend will again be considered.
+
+    Parameters
+    ----------
+    backend : {object, 'scipy'}
+        The backend to skip.
+        Can either be a ``str`` containing the name of a known backend
+        {'scipy'} or an object that implements the uarray protocol.
+
+    Notes
+    -----
+    .. versionadded:: 1.8.0
+
+    Examples
+    --------
+    >>> import scipy.ndimage as ndimage
+    >>> # Calls default SciPy backend
+    >>> ndimage.correlate(np.arange(10), [1, 2.5])
+    array([ 0,  2,  6,  9, 13, 16, 20, 23, 27, 30])
+    >>> # Explicitly skip SciPy backend, leaving no implementation available
+    >>> with ndimage.skip_backend('scipy'):
+    ...     ndimage.correlate(np.arange(10), [1, 2.5])
+    Traceback (most recent call last):
+        ...
+    BackendNotImplementedError: No selected backends had an implementation ...
+    """
+    backend = _backend_from_arg(backend)
+    return ua.skip_backend(backend)
+
+
+set_global_backend('scipy')
```

Comment:
Are there no tests passing lists? I would expect that to fail without setting `coerce=True`.
```suggestion
set_global_backend('scipy', coerce=True)
```

Reply:
`coerce=True` is a default in ndimage `set_global_backend`. See [LOC100](https://github.com/scipy/scipy/blob/7c4ce13a58a12b01fa0a3b1fe0aeb09db92df8f4/scipy/ndimage/_backend.py#L100) in `_backend.py`.

Reply:
There are definitely a few tests that provide lists as input.